serde_codegen/src/attr.rs

```
@@ -234,107 +290,99 @@ impl FieldAttrs {
             Some(ident) => ident,
-            None => builder.id(index.to_string()),
-        };
-
-        let mut field_attrs = FieldAttrs {
-            name: Name::new(field_ident),
-            skip_serializing_field: false,
-            skip_deserializing_field: false,
-            skip_serializing_if: None,
-            default: FieldDefault::None,
-            serialize_with: None,
-            deserialize_with: None,
-            ser_bound: None,
-            de_bound: None,
+            None => index.to_string().to_ident(),
         };
 
         for meta_items in field.attrs.iter().filter_map(get_serde_meta_items) {
             for meta_item in meta_items {
+                let span = meta_item.span;
                 match meta_item.node {
                     // Parse `#[serde(rename="foo")]`
                     ast::MetaItemKind::NameValue(ref name, ref lit) if name == &"rename" => {
-                        let s = try!(get_str_from_lit(cx, name, lit));
-                        field_attrs.name.serialize_name = Some(s.clone());
-                        field_attrs.name.deserialize_name = Some(s);
+                        if let Ok(s) = get_str_from_lit(cx, name, lit) {
+                            ser_name.set(span, s.clone());
+                            de_name.set(span, s);
+                        }
                     }
 
                     // Parse `#[serde(rename(serialize="foo", deserialize="bar"))]`
                     ast::MetaItemKind::List(ref name, ref meta_items) if name == &"rename" => {
-                        let (ser_name, de_name) = try!(get_renames(cx, meta_items));
-                        if ser_name.is_some() {
-                            field_attrs.name.serialize_name = ser_name;
-                        }
-                        if de_name.is_some() {
-                            field_attrs.name.deserialize_name = de_name;
+                        if let Ok((ser, de)) = get_renames(cx, meta_items) {
```

Yes I am. This is on purpose. The span_err has already been emitted inside get_ser_and_de. Ignoring the error here means that we can simultaneously show errors for all of the bad attributes in a file, rather than showing them one at a time. Since there were errors, compilation will fail after the macro expansion phase.